lib/oli_web/live/products/payments/discounts/show_view.ex

```diff
@@ -125,10 +140,9 @@ defmodule OliWeb.Products.Payments.Discounts do
   def handle_event("save", %{"discount" => params}, socket) do
     socket = clear_flash(socket)
 
-    rels_params = get_rels_params(socket.assigns)
     attrs = %{
-      section_id: rels_params.section_id,
-      institution_id: rels_params.institution_id,
+      section_id: (if socket.assigns.live_action == :institution, do: nil, else: socket.assigns.product.id),
+      institution_id: (if socket.assigns.live_action == :product_new, do: get_institution_id(params["institution_id"]), else: socket.assigns.institution.id),
       percentage: params["percentage"],
       amount: params["amount"],
       type: params["type"]
```

Comment:
This appears to not have been part of this code change, but it is a common pattern in the system to be redirected back to the table view with the successful flash message rendered after save. I cannot think of a scenario where someone would want to remain on this page after saving, and in fact this might confuse some users who still see the form and think that maybe the save wasn't successful.

Because this wasn't part of the original ticket here and it's an admin only functionality, it probably okay to leave it as is for this PR but I'll leave it up to you if you want to change this or not.

Reply:
Makes total sense to me, it's not a big change so I can include it without much effort in favor of the consistency throughout the app. 

I'll wait to see if James has some feedback and will do all at once. Thanks !